pkg/nuctl/test/get_test.go

```diff
@@ -45,6 +45,93 @@ func (suite *GetTestSuite) SetupSuite() {
 	})
 }
 
+func (suite *GetTestSuite) TestMultipleGet() {
+	numOfFunctions := 3
+	var functionNames []string
+
+	// initialize functions
+	for functionIdx := 0; functionIdx < numOfFunctions; functionIdx++ {
+		uniqueSuffix := fmt.Sprintf("-%s-%d", xid.New().String(), functionIdx)
+		imageName := "nuclio/deploy-test" + uniqueSuffix
+		functionName := "reverser" + uniqueSuffix
+
+		// add function name to list
+		functionNames = append(functionNames, functionName)
+		namedArgs := map[string]string{
+			"path":    path.Join(suite.GetFunctionsDir(), "common", "reverser", "golang"),
+			"image":   imageName,
+			"runtime": "golang",
+			"handler": "main:Reverse",
+		}
+
+		err := suite.ExecuteNutcl([]string{
+			"deploy",
+			functionName,
+			"--verbose",
+			"--no-pull",
+		}, namedArgs)
+		suite.Require().NoError(err)
+
+		// cleanup
+		defer func() {
+
+			// make sure to clean up after the test
+			suite.dockerClient.RemoveImage(imageName)
+		}()
+	}
+
+	// number of combinations need to be check
+	tests := [][]string{{functionNames[0], functionNames[1], functionNames[2]}, {functionNames[1], functionNames[1]},
+		{functionNames[0], functionNames[1]}}
+
+	// Iterate over tests and check for right results for each one
+	for testIndex, test := range tests {
+		suite.outputBuffer.Reset()
+		err := suite.ExecuteNutcl(append([]string{"get", "function"}, test...), nil)
+		suite.Require().NoError(err)
+		foundFunctions := make([]bool, len(test))
+
+		// iterate over all lines in get result. for each function created in this test that we find,
+		// set the equivalent boolean in foundFunctions
+		scanner := bufio.NewScanner(&suite.outputBuffer)
+	searchResultsLabel:
+		for scanner.Scan() {
+			for functionIdx, functionName := range test {
+
+				// if the function name is in the list, remove it
+				if strings.Contains(scanner.Text(), functionName) {
+
+					// check for double result
+					if foundFunctions[functionIdx] == true {
+						foundFunctions[functionIdx] = false
+						break searchResultsLabel
+					} else {
+						foundFunctions[functionIdx] = true
+						break
+					}
+				}
+			}
+		}
+
+		for _, foundFunction := range foundFunctions {
+
+			// unique behaviour of double function test leaded to this intervention
```

Comment:
I don't think it's acceptable to do this. If your way of marking found functions isn't able to deal with the same function appearing twice (which is wrong behavior in itself), just change that way. Instead of tracking _whether_ a function appeared, track _how many times_ it appeared. Each unique function should appear once.

Keep in mind the duplicates shouldn't be removed at the renderer, but at the root of the GetFunctions implementation - there is no valid reason for GetFunctions to return the same function twice, ever. The simplest way sounds like checking for dupes at the viper getter (rather than in both platform implementations)